### DISCUSSION TITLE
Discussions enabled

### DISCUSSION THREAD

**Discussion by lexelby** [inkstitch/inkstitch#930] [General]
Yay, we can finally get rid of the old Discussion Forum issue!

**Comment by kaalleen** [comment]
Let's move it over to here :)

**Comment by JoSeaMor3** [comment]
I'm new here, new to Inkscape/Inkstitch, AND Embroidery! I'm also impatient and I'm trying to learn way too much at once. I just want to make something not boring. I just bought an entry sewing/embroidery combo unit to start with and see how I do. I found some great Youtube videos of users on inkscape. What is your favorite video/tutorial that you learned the most from? This guy has some easy beginner pointers. https://www.youtube.com/channel/UCZ4uSYhY8rHsuKG7lsBxDkQ/videos [](url)

**Comment by ghost** [comment]
Hello all, I have an interesting question to pose. I am trying to put together an image processing application that will take a jpg and create an embroidery file from it.

I have hit a wall though when it comes to the file format I hope to write it to (.JEF) I can find next to no documentation online that outlines the file structure etc. I have used ink stitch quite successfully in the past and notice that the .JEF format is an option.

If at all possible, would anyone be able to point me in the direction of some information about how to write the JEF format 

All I have found so far is this one listing on the following website: 
https://community.kde.org/Projects/Liberty/File_Formats/Janome_Embroidery_Format#File-header

**Reply by lexelby** [reply]
Ink/Stitch uses this library under the hood: https://github.com/inkstitch/pyembroidery

There's not much documentation out there about embroidery file formats.  Some of the formats in pyembroidery have issues, but I can definitely attest that JEF works properly because I used to have a Janome 400e.  I hope that helps, and I'd be interested to see your program, if you care to share.

**Reply by ghost** [reply]
@lexelby 

I'll take a look at that thank you. Its not really passed the image processing stage yet. 

Its a final year project for my degree. I have learnt python specifially for this project and it has been a bit challenging but yes certainly when I get it finished in will get you a link to the github. I do think I'll have a lot of bugs to fix and areas to clear up after I have submitted it for my course.

TL:DR - New to python don't expect a masterpiece haha

**Reply by JoshuaMcroberts** [reply]
Hey @lexelby I only realised after I posted the above that I was signed into the wrong Git account. I will send you an invite to the repo for the above-mentioned project as it is linked to this account. 

If possible, I would also like to find out the data and format need to feed into the writer_encoder.py. 

I am at the stage of creating the shape objects to be converted into paths.

Any advice or assistance in understanding the pyembroidery library would be invaluable at to me at this stage. 

Thank for the help so far.
